accounts/url_test.go

```diff
@@ -36,6 +36,11 @@ func TestURLParsing(t *testing.T) {
 	if err == nil {
 		t.Error("expected err, got: nil")
 	}
+	
+	_, err = parseURL("")
+	if err == nil {
+		t.Error("expected err, got: nil")
+	}
```

Comment:
An alternative way to do this which is a bit more easily extendable would be this: 
```golang
	for _, u := range []string{"ethereum.org", ""}{
		if _, err = parseURL(u); err == nil {
			t.Errorf("input %v, expected err, got: nil", u)
		}
	}
```
As it's a kind of small test though, I don't have a problem with the way you've done it

Reply:
appreciate it.
It's applied.

Reply:
I think null string is incorrect value for test it.
```go
parts := strings.Split(url, "://")
if len(parts) != 2 || parts[0] == "" {
		return URL{}, errors.New("protocol scheme missing")
}
```
cause, ""  split value's length is not 2...
If check the logic,
```go
parts[0] == "" 
```
I think, It needs "://"

Reply:
Null string is incorrect value for test it,
```go
if len(parts) != 2 || parts[0] == ""
// parts[0] == "" <- this
```
<br>
If test this logic, I think it need

```go
"://"
```

<br>

```go
url := ""
parts := strings.Split(url, "://")
// len(parts): 1 

url = "://"
parts = strings.Split(url, "://")
// len(parts): 2
// parts[0] : "", parts[1]: ""
```

<br>

I'm not sure is this necessary?
I think it's enough this
```go
if len(parts) != 2
```

Reply:
```go
if len(parts) != 2 || parts[0] == ""
```

```go
// parts[0] == "" is this necessary?

parts := strings.split(url, "://")
// "ethereum.org"
// ""
// length: 1
```

Reply:
That's because `parseURL("")` does not test anything more than `parseURL("ethereum.org")` already tested.

It does not test the `|| parts[0] == ""` branch that you intended. It would need to be something like  `parseURL("://ethereum.org")`.